packages/fuels-test-helpers/src/lib.rs

```diff
@@ -165,17 +165,25 @@ pub async fn setup_test_client(
     let message_configs = get_message_configs(messages);
 
     // Setup node config with genesis coins and utxo_validation enabled
-    let chain_conf = chain_config.unwrap_or_else(|| ChainConfig {
+    let mut chain_conf = chain_config.unwrap_or_else(|| ChainConfig {
         initial_state: Some(StateConfig {
             coins: Some(coin_configs),
             contracts: None,
             messages: Some(message_configs),
             ..StateConfig::default()
         }),
-        transaction_parameters: consensus_parameters_config.unwrap_or_default(),
         ..ChainConfig::local_testnet()
     });
 
+    chain_conf = if consensus_parameters_config.is_some() {
+        ChainConfig {
+            transaction_parameters: consensus_parameters_config.unwrap(),
+            ..chain_conf
+        }
+    } else {
+        chain_conf
+    };
+
```

Comment:
```rust
    let chain_conf = chain_config.unwrap_or_else(|| ChainConfig {
        transaction_parameters: consensus_parameters_config.unwrap_or_default(),
        initial_state: Some(StateConfig {
            coins: Some(coin_configs),
            contracts: None,
            messages: Some(message_configs),
            ..StateConfig::default()
        }),
        ..ChainConfig::local_testnet()
    });
```

Reply:
> ```rust
>     let chain_conf = chain_config.unwrap_or_else(|| ChainConfig {
>         transaction_parameters: consensus_parameters_config.unwrap_or_default(),
>         initial_state: Some(StateConfig {
>             coins: Some(coin_configs),
>             contracts: None,
>             messages: Some(message_configs),
>             ..StateConfig::default()
>         }),
>         ..ChainConfig::local_testnet()
>     });
> ```

It can't be like this. This was exactly what was causing the bug.

Reply:
Good catch!